packages/server/lib/controllers/connection.controller.ts

```diff
@@ -61,29 +62,30 @@ class ConnectionController {
                 environment_id: environment.id
             };
 
-            const { success, error, response: connection } = await connectionService.getConnection(connectionId, providerConfigKey, environment.id);
-
-            if (!success) {
-                errorManager.errResFromNangoErr(res, error);
+            let success: boolean;
+            let error: NangoError | null = null;
+            let connection: Connection | null = null;
+            if (instantRefresh) {
+                const credentialResponse = await connectionService.getConnectionCredentials({
+                    account,
+                    environment,
+                    connectionId,
+                    providerConfigKey,
+                    logContextGetter,
+                    instantRefresh: false
+                });
+                success = credentialResponse.isOk();
 
-                return;
+                if (credentialResponse.isOk()) {
+                    connection = credentialResponse.value;
+                } else {
+                    error = credentialResponse.error;
+                }
+            } else {
+                ({ success, error, response: connection } = await connectionService.getConnection(connectionId, providerConfigKey, environment.id));
```

Comment:
you refactored `getConnectionCredentials` to return a `Result`. Might be worthwhile to also refactor `getConnection`. I think that would facilitate the error handling here since you wouldn't need a success or error temp variable